Source/Tools/FEXLoader/LinuxSyscalls/SyscallsSMCTracking.cpp

```diff
@@ -163,33 +165,34 @@ void SyscallHandler::MarkGuestExecutableRange(uint64_t Start, uint64_t Length) {
 }
 
 // Used for AOT
-FEXCore::HLE::AOTIRCacheEntryLookupResult SyscallHandler::LookupAOTIRCacheEntry(uint64_t GuestAddr) {
-  FHU::ScopedSignalMaskWithSharedLock lk(_SyscallHandler->VMATracking.Mutex);
-  auto rv = FEXCore::HLE::AOTIRCacheEntryLookupResult(nullptr, 0);
+FEXCore::HLE::AOTIRCacheEntryLookupResult SyscallHandler::LookupAOTIRCacheEntry(FEXCore::Core::InternalThreadState *Thread, uint64_t GuestAddr) {
+  FEXCore::ScopedDeferredSignalWithSharedLock lk(VMATracking.Mutex, Thread);
 
   // Get the first mapping after GuestAddr, or end
   // GuestAddr is inclusive
   // If the write spans two pages, they will be flushed one at a time (generating two faults)
-  auto Entry = _SyscallHandler->VMATracking.LookupVMAUnsafe(GuestAddr);
-
-  if (Entry != _SyscallHandler->VMATracking.VMAs.end()) {
-    rv.Entry = Entry->second.Resource ? Entry->second.Resource->AOTIRCacheEntry : nullptr;
-    rv.VAFileStart = Entry->second.Base - Entry->second.Offset;
+  auto Entry = VMATracking.LookupVMAUnsafe(GuestAddr);
+  if (Entry == VMATracking.VMAs.end()) {
+    return {nullptr, 0};
   }
 
-  return rv;
+  return {
+    Entry->second.Resource ? Entry->second.Resource->AOTIRCacheEntry : nullptr,
+    Entry->second.Base - Entry->second.Offset
+  };
 }
 
 // MMan Tracking
-void SyscallHandler::TrackMmap(uintptr_t Base, uintptr_t Size, int Prot, int Flags, int fd, off_t Offset) {
-	Size = FEXCore::AlignUp(Size, FHU::FEX_PAGE_SIZE);
+void SyscallHandler::TrackMmap(FEXCore::Core::InternalThreadState *Thread, uintptr_t Base, uintptr_t Size, int Prot, int Flags, int fd, off_t Offset) {
+  Size = FEXCore::AlignUp(Size, FHU::FEX_PAGE_SIZE);
 
   if (Flags & MAP_SHARED) {
     CTX->MarkMemoryShared();
   }
 
   {
-    FHU::ScopedSignalMaskWithUniqueLock lk(_SyscallHandler->VMATracking.Mutex);
+    // Frontend calls this with nullptr Thread.
```

Comment:
It's a comment as to why `ScopedPotentialDeferredSignalWithUniqueLock` is used since the frontend doesn't have a valid thread object early in initialization.
Added a comment here.

Reply:
That's why I changed the comment to:
```
    // Frontend calls this with nullptr Thread during initialization.
    // This is why `ScopedPotentialDeferredSignalWithUniqueLock` is used here.
    // To be more optimal the frontend should provide this code with a valid Thread object earlier.
```
So opposed to getting an internal thread object sooner in initialization so it doesn't need to use the `ScopedPotential` and instead just `Scoped` variant of the unique lock.

Reply:
Ah, I see. Suggested a rephrase in the diff.